### PR TITLE
ENH/TST: add makedirs with mode fix, make_dir_tree

### LINKED ISSUE #84: Put Stuart's directory-generating script into ophyd.utils

### PR DESCRIPTION
**Pull request by klauer**
Closes #84

### CODE REVIEW

File: ophyd/utils/paths.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    '''
+    # Permissions not working with os.makedirs -
+    # See: http://stackoverflow.com/questions/5231901
+    if not path or os.path.exists(path):
+        return []
+
+    head, tail = os.path.split(path)
+    ret = makedirs(head, mode=mode, mode_base=mode_base)
+    try:
+        os.mkdir(path)
```
Comment (tacaswell) [inline ophyd/utils/paths.py]: you can use the `exists_ok` kwarg here to avoid this try/except.

Comment (klauer) [inline ophyd/utils/paths.py]: `exists_ok` is in `os.makedirs`, which IIRC didn't set permissions correctly, hence this function. Though not sure if this was only necessary in py2k...

Comment (tacaswell) [inline ophyd/utils/paths.py]: that is what the `os.chmod` a few lines down fixes.  I think you can let python do the recursion checking for you and then juts fix up the permissions?

Comment (klauer) [inline ophyd/utils/paths.py]: Fair enough, it would remove this ugly exception handling at least. Will fix.

**Comment by klauer** [comment]
Rebased to get those test fixes from other PRs and modified to use `os.makedirs`.

**Pull request merged**